ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/B11rSubCommand.java

```diff
@@ -0,0 +1,245 @@
+  }
+
+  @SuppressWarnings("unused")
+  public B11rSubCommand(final EvmToolCommand parentCommand) {
+    // PicoCLI requires this too
+    this.parentCommand = parentCommand;
+  }
+
+  @Override
+  public void run() {
+    objectMapper.setDefaultPrettyPrinter(
+        (new DefaultPrettyPrinter())
+            .withSpacesInObjectEntries()
+            .withObjectIndenter(DefaultIndenter.SYSTEM_LINEFEED_INSTANCE.withIndent("  "))
+            .withArrayIndenter(DefaultIndenter.SYSTEM_LINEFEED_INSTANCE.withIndent("  ")));
+    final ObjectReader t8nReader = objectMapper.reader();
+    objectMapper.disable(Feature.AUTO_CLOSE_SOURCE);
+
+    ObjectNode config;
+    try {
+      if (header.equals(stdinPath)
+          || txs.equals(stdinPath)
+          || ommers.equals(stdinPath)
+          || sealClique.equals(stdinPath)) {
+        config =
+            (ObjectNode)
+                t8nReader.readTree(new InputStreamReader(parentCommand.in, StandardCharsets.UTF_8));
+      } else {
+        config = objectMapper.createObjectNode();
+      }
+
+      if (!header.equals(stdinPath)) {
+        config.set(
+            "header", t8nReader.readTree(new FileReader(header.toFile(), StandardCharsets.UTF_8)));
+      }
+      if (!txs.equals(stdinPath)) {
+        try (FileReader reader = new FileReader(txs.toFile(), StandardCharsets.UTF_8)) {
+          config.set("txs", t8nReader.readTree(reader));
+        }
+      }
+      if (!ommers.equals(stdinPath)) {
+        try (FileReader reader = new FileReader(ommers.toFile(), StandardCharsets.UTF_8)) {
+          config.set("ommers", t8nReader.readTree(reader));
+        }
+      }
+      if (!sealClique.equals(stdinPath)) {
+        try (FileReader reader = new FileReader(sealClique.toFile(), StandardCharsets.UTF_8)) {
+          config.set("clique", t8nReader.readTree(reader));
+        }
+      }
+    } catch (final JsonProcessingException jpe) {
+      parentCommand.out.println("File content error: " + jpe);
+      jpe.printStackTrace();
+      return;
+    } catch (final IOException e) {
+      LOG.error("Unable to read state file", e);
+      return;
+    }
+
+    var testHeader = this.readHeader(config.get("header"));
+    Bytes txsBytes = null;
+
+    if (config.has("txs")) {
+      String txsString = config.get("txs").textValue();
+      if (txsString.length() > 0) {
+        txsBytes = Bytes.fromHexString(txsString);
+      }
+    }
+
+    var newHeader =
+        BlockHeaderBuilder.fromHeader(testHeader)
+            .blockHeaderFunctions(new MainnetBlockHeaderFunctions())
+            .buildBlockHeader();
+    BytesValueRLPOutput rlpOut = new BytesValueRLPOutput();
+    rlpOut.startList();
+    newHeader.writeTo(rlpOut);
+    if (txsBytes != null && txsBytes.size() > 0) {
+      rlpOut.writeRaw(txsBytes);
+    } else {
+      rlpOut.startList();
+      rlpOut.endList();
+    }
+    // ommers
+    rlpOut.startList();
+    rlpOut.endList();
+
+    // withdrawals
+    // TODO
```

Comment:
Still TODO? are you blocked on something related to withdrawals?

Reply:
Waiting on the `execution-spec-tests` to pass in withdrawals information to the `b11r` tool.  When that support comes this is where it lands.  clarified todo.